pkg/sdk/grants_impl.go

```diff
@@ -33,6 +65,37 @@ func (v *grants) RevokePrivilegesFromAccountRole(ctx context.Context, privileges
 	opts.on = on
 	opts.accountRole = role
 	logging.DebugLogger.Printf("[DEBUG] Revoke privileges from account role: opts %+v", opts)
+
+	// Snowflake doesn't allow bulk operations on Pipes. Because of that, when SDK user
+	// issues "revoke x on all pipes" operation, we'll go and revoke specified privileges
+	// from every Pipe one by one.
+	if on != nil &&
+		on.SchemaObject != nil &&
+		on.SchemaObject.All != nil &&
+		on.SchemaObject.All.PluralObjectType == PluralObjectTypePipes {
+		return v.runOnAllPipes(
+			ctx,
+			on.SchemaObject.All.InDatabase,
+			on.SchemaObject.All.InSchema,
+			func(pipe Pipe) error {
+				return v.client.Grants.RevokePrivilegesFromAccountRole(
```

Comment:
it seems like a few things could be extracted here: condition, all parameters are the same, the only diff is `GrantPrivilegesToAccountRole` vs `RevokePrivilegesFromAccountRole`, right?

Reply:
And also the following ones

Reply:
I tried to extract it, but in condition `on` is used and it's different between account and database roles, InDatabase and InSchema I was able to extract and just pass it to the method (same as previous `on` is different) and the SDK call had to be exposed, because pass-through parameters are also different between functions (like privileges, role and opts).